### PR TITLE
Feature/backend server connection status

### PR DESCRIPTION
**Pull request by ShobhitAd**
### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [x] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).

#### Core Tests
Start HMI before starting backend server
Click the connection status icon to update the Backend server URL and click OK to attempt re-connection

Core version / branch / commit hash / module tested against: [INSERT]
Proxy+Test App name / version / branch / commit hash / module tested against: [INSERT]

### Summary
Adds connection status icon as well as the ability to reconnect to the backend server (or connect to a different backend server)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by iCollin** [comment]
Should probably be added to this PR: https://github.com/smartdevicelink/generic_hmi/pull/482/commits/0327ef0aa2a8281388eb210cec4f0ed1ebda8b46

**Comment by iCollin** [comment]
Something to consider: the PTU process currently requires the backend server shares a file system with the HMI.

File: src/js/AppHeader.js [hunk 1]
```diff
@@ -17,19 +18,52 @@ import ScrollableMessage from './ScrollableMessage';
...
     }
 }
 
+class BackendConnectionStatus extends React.Component {
+    render() {
+        let statusIcon = FileSystemController.isConnected() ? <IconFSCConnected/>
+            : <IconFSCDisconnected/>;
+        let attemptReconnection = (url) => {
+            window.flags.FileSystemApiUrl = url;
+            localStorage.setItem("FileSystemApiUrl", window.flags.FileSystemApiUrl)
```
Comment (iCollin) [inline src/js/AppHeader.js:42]: I would suggest only saving `FileSystemApiUrl` in local storage if the connection is successful, maybe just moving this line into the connect resolve method.

**Review by iCollin — COMMENTED** [review]
It looks like the diff on this PR didn't update when I merged the `consistent_body_margins` branch into develop, so just FYI I did not review any `scss` files besides `src/css/components/_app-header.scss`

File: src/js/AppHeader.js (hunk 1 above)
Comment (ShobhitAd) [reply src/js/AppHeader.js]: Good idea. Changed in https://github.com/smartdevicelink/generic_hmi/pull/481/commits/67dd0c091ee8cfd7ffdfc063eaecd70c8693cdf9

**Review by iCollin — APPROVED** [review]

**Pull request merged**